### ISSUE TITLE
Buffer modified icon in newer :ui modeline

### ISSUE THREAD

**Issue by filalex77** [hlissner/doom-emacs#1216]
### Observed behavior

I don't this it should look like this.

![image](https://user-images.githubusercontent.com/12615679/53790715-492a4d80-3f30-11e9-9e53-bec988bf9174.png)

This happens both on my [fork of doom-emacs](https://gitlab.com/BrightOne/doom-emacs), as well as on upstream doom-emacs without my private customisations.

### Expected behavior

This segment looking like this:

![image](https://raw.githubusercontent.com/hlissner/doom-emacs/screenshots/tabs.png)
### Steps to reproduce

1. Update to latest `develop` (for example, c1beb39c5b81f97e49513ebf765ee505fec922fe)
2. `make clean install update autoloads`
3. Launch Emacs
4. Make changes in buffer

### System information

<details>
<summary>Click to expand</summary>

```
Doom Doctor
Emacs v26.1
Doom v2.0.9 (d480e9ae)
shell: /bin/zsh
Compiled with:
  XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS NOTIFY ACL GNUTLS LIBXML2
  FREETYPE XFT ZLIB GTK3 X11 MODULES THREADS LCMS2
uname -a:
  Linux thinktoo 4.20.10-gentoo #1 SMP Sat Feb 16 12:54:41 EET 2019 x86_64
  Intel(R) Core(TM) i5-2520M CPU @ 2.50GHz GenuineIntel GNU/Linux

Checking your OS...
Checking your fonts...
Checking gnutls/openssl...
Testing your root certificates...
Checking for GNU/BSD tar...
DOOM Initialized evil-collection-calendar [feature/evil]
DOOM Initialized evil-collection-comint [feature/evil]
DOOM Initialized evil-collection-custom [feature/evil]
DOOM Initialized evil-collection-help [feature/evil]
DOOM Initialized evil-collection-dired [feature/evil]
DOOM Initialized evil-collection-doc-view [feature/evil]
DOOM Initialized evil-collection-epa [feature/evil]
DOOM Initialized evil-collection-mu4e [feature/evil]
DOOM Initialized evil-collection-outline [feature/evil]
DOOM Initialized evil-collection-diff-mode [feature/evil]
DOOM Initialized evil-collection-magit [feature/evil]
DOOM Initialized evil-collection-compile [feature/evil]
DOOM Initialized evil-collection-grep [feature/evil]
DOOM Initialized evil-collection-magit-todos [feature/evil]
DOOM Running doom hook: evil-mode
DOOM Running doom hook: +evil|init-popup-rules
DOOM Running doom hook: +wakatime|delayed-autostart
DOOM Running doom hook: +workspaces|init
DOOM Initialized evil-collection-info [feature/evil]
Checking your enabled modules...
Loading /home/brightone/.emacs.d.doom/core/autoload/cache.el (source)...
  > (:lang cc) Couldn’t find the rtag client and/or server programs
    (rdm). Disabling rtags support
  > (:app write) Couldn’t find languagetool-commandline.jar


There are 2 warnings!
No RTags process running (rdm)...
```
</details>

**Comment by hlissner** [comment]
I am unable to reproduce this, but I've heard one other report matching it. I don't know the cause. 

@seagle0128 would you know what's causing this?

**Comment by seagle0128** [comment]
It should be related to the font settings. I guess @filalex77 is using `Fira Code`, right?

@hlissner In doom-emacs, `+pretty-code-fira-code-font-ligatures` defines some symbols, which are conflict with `all-the-icons`. In this case, is `material` fonts. Refer to https://github.com/seagle0128/doom-modeline/issues/69 .

**Comment by filalex77** [comment]
@seagle0128 Thanks, that seems to be the cause.

**Comment by hlissner** [comment]
Just an fyi, I've merged the fix for this in 909c277.
